addons/mail/tests/template_statusbar_time_testing.py

```diff
@@ -0,0 +1,87 @@
+from datetime import datetime
+from unittest.mock import patch
+
+from odoo.addons.mail.tests.common import MockStatusbarTime
+
+
+class TestStatusbarTimeMixin(MockStatusbarTime):
+
+    def setUp(self, tested_model, stage_model, model_fields=None, stage_fields=None):
+
+        stage_1_values = {'name': 'Stage 1'}
+        stage_2_values = {'name': 'Stage 2'}
+        stage_3_values = {'name': 'Stage 3'}
+        stage_4_values = {'name': 'Stage 4'}
+        if stage_fields:
+            for stage_value in [stage_1_values, stage_2_values, stage_3_values, stage_4_values]:
+                stage_value.update(stage_fields)
+
+        self.stage_1 = self.env[stage_model].create(stage_1_values)
+        self.stage_2 = self.env[stage_model].create(stage_2_values)
+        self.stage_3 = self.env[stage_model].create(stage_3_values)
+        self.stage_4 = self.env[stage_model].create(stage_4_values)
+
+        record_values = {'name': 'test record', 'stage_id': self.stage_1.id}
+        if model_fields:
+            record_values.update(model_fields)
+
+        self.mock_start_time = datetime(2023, 2, 15, 12, 0, 0)
+
+        with patch.object(self.env.cr, 'now', return_value=self.mock_start_time):
+            self.rec_1, self.rec_2, self.rec_3, self.rec_4 = self.env[tested_model].create(
+                [record_values for i in range(4)])
+        self.flush_tracking()
+
+    def template_test_record_json_status_time(self):
```

Comment:
this "template" naming is confusing for people working in mail like me, we expect something with (mail) templates :D why not simply private methods like "_test_record_json_status_time" ?